docs/csharp/language-reference/operators/boolean-logical-operators.md

```diff
@@ -0,0 +1,159 @@
+
+The [logical AND operator](#logical-and-operator-) `&` also computes the logical AND of its operands, but always evaluates both operands.
+
+## Conditional logical OR operator ||
+
+The conditional logical OR operator `||`, also known as the "short-circuiting" logical OR operator, computes the logical OR of its operands. The result of `x || y` is `true` if either `x` or `y` evaluates to `true`. Otherwise, the result is `false`. If the first operand evaluates to `true`, the second operand is not evaluated and the result of operation is `true`. The following example demonstrates that behavior:
+
+[!code-csharp-interactive[conditional logical OR](~/samples/snippets/csharp/language-reference/operators/LogicalOperators.cs#ConditionalOr)]
+
+The [logical OR operator](#logical-or-operator-) `|` also computes the logical OR of its operands, but always evaluates both operands.
+
+## Compound assignment
+
+For a binary operator `op`, a compound assignment expression of the form
+
+```csharp
+x op= y
+```
+
+is equivalent to
+
+```csharp
+x = x op y
+```
+
+except that `x` is only evaluated once.
+
+The `&`, `|`, and `^` operators support compound assignment, as the following example shows:
+
+[!code-csharp-interactive[compound assignment](~/samples/snippets/csharp/language-reference/operators/LogicalOperators.cs#CompoundAssignment)]
+
+The conditional logical operators `&&` and `||` don't support compound assignment.
+
+## Operator precedence
+
+The following list orders logical operators starting from the highest precedence to the lowest:
+
+- Logical negation operator `!`
+- Logical AND operator `&`
+- Logical exclusive OR operator `^`
+- Logical OR operator `|`
+- Conditional logical AND operator `&&`
+- Conditional logical OR operator `||`
+
+Use parentheses, `()`, to change the order of evaluation imposed by operator precedence:
+
+[!code-csharp-interactive[operator precedence](~/samples/snippets/csharp/language-reference/operators/LogicalOperators.cs#Precedence)]
+
+For the complete list of C# operators ordered by precedence level, see [C# operators](index.md).
+
+## Operator overloadability
+
+A user-defined type can [overload](../keywords/operator.md) the `!`, `&`, `|`, and `^` operators. When a binary operator is overloaded, the corresponding compound assignment operator is also implicitly overloaded. A user-defined type cannot explicitly overload a compound assignment operator.
+
+A user-defined type cannot overload the conditional logical operators `&&` and `||`. However, if a user-defined type overloads the `&` operator (the `|` operator) and the [true and false operators](../keywords/true-false-operators.md) in a certain way, the `&&` operation (the `||` operation) can be evaluated for the operands of that type. For more information, see the [User-defined conditional logical operators](~/_csharplang/spec/expressions.md#user-defined-conditional-logical-operators) section of the [C# language specification](~/_csharplang/spec/introduction.md).
```

Comment:
The parenthetical expressions "(the `|` operator)" and "(the `||` operator) look odd to me. I don't have a good suggestion right now. I'll keep noodling. But, if you come up with a better wording make an edit here.

Reply:
I like you've mentioned it. I'm also not fully satisfied with parentheses. Wander more about it, then.

Reply:
Rephrased.